### PR TITLE
autolock with no device security

### LINKED ISSUE #318: Default Auto Lock timer to "Never" when device security is not set up
When a user does not have device security set up on their device, we need to default the Auto Lock timer setting to "Never".

When a user taps on the Auto Lock timer setting, display a dialog to the user informing them in order to update this setting they need to set up device security on their phone.

### Visual Design

**Zeplin**
https://zpl.io/bPv9kgl

### PR DESCRIPTION
**Pull request by sashei**
Fixes #318 

## Testing and Review Notes
**code reviewers**: I snuck in a refactor of how we're doing system dialogs here to remove some code bloat from the `RoutePresenter`.

**functional reviewers**: The expected behavior when enabling and disabling device security is that the app will change to the Five Minutes and Never defaults for those two cases, respectively.

## To Do

- [x] add “WIP” to the PR title if pushing up but not complete nor ready for review
- [x] double check the original issue to confirm it is fully satisfied
- [x] add testing notes and screenshots in PR description to help guide reviewers
- [x] add unit tests
- [x] request the "UX" team perform a design review (if/when applicable)
- [x] make sure CI builds are passing (e.g.: fix lint and other errors)

### CODE REVIEW

File: app/src/androidTest/java/mozilla/lockbox/robots/FingerprintDialogRobot.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ import br.com.concretesolutions.kappuccino.actions.ClickActions.click
 import br.com.concretesolutions.kappuccino.assertions.VisibilityAssertions.displayed
 import mozilla.lockbox.R
 
-// Fingerprint Dialog
+// Fingerprint DialogAction
```
Comment (eliserichards) [inline app/src/androidTest/java/mozilla/lockbox/robots/FingerprintDialogRobot.kt]: why renamed?

File: app/src/androidTest/java/mozilla/lockbox/robots/UIComponentRobot.kt [hunk 2]
```diff
@@ -10,7 +10,7 @@ import br.com.concretesolutions.kappuccino.actions.ClickActions.click
 import br.com.concretesolutions.kappuccino.assertions.VisibilityAssertions.displayed
 import mozilla.lockbox.R
 
-// Fingerprint Dialog
+// Fingerprint DialogAction
```
Comment (eliserichards) [inline app/src/androidTest/java/mozilla/lockbox/robots/UIComponentRobot.kt]: why renamed?

File: app/src/main/java/mozilla/lockbox/action/RouteAction.kt [hunk 3]
```diff
@@ -10,11 +10,9 @@ import android.net.Uri
 import android.os.Build
 import androidx.annotation.RequiresApi
 import androidx.annotation.StringRes
-import mozilla.lockbox.R
 import mozilla.lockbox.flux.Action
-import mozilla.lockbox.support.Constant
 
-sealed class RouteAction(
+open class RouteAction(
```
Comment (eliserichards) [inline app/src/main/java/mozilla/lockbox/action/RouteAction.kt]: does this have to be open in order for the `DialogAction` to be able to see it?

File: app/src/main/java/mozilla/lockbox/model/DialogViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.model
+
+import androidx.annotation.ColorRes
+import androidx.annotation.StringRes
+
+data class DialogViewModel(
```
Comment (eliserichards) [inline app/src/main/java/mozilla/lockbox/model/DialogViewModel.kt]: Love this

**Comment by sashei** [comment]
I have no idea why the `LockedPresenter` test is failing on CI; it's not doing that locally for me :(

File: app/src/main/java/mozilla/lockbox/presenter/SettingPresenter.kt [hunk 5]
```diff
@@ -47,7 +49,11 @@ class SettingPresenter(
 
     private val autoLockTimeClickListener: Consumer<Unit>
         get() = Consumer {
-            dispatcher.dispatch(RouteAction.AutoLockSetting)
+            if (fingerprintStore.isDeviceSecure) {
+                dispatcher.dispatch(RouteAction.AutoLockSetting)
+            } else {
+                dispatcher.dispatch(DialogAction.SecurityDisclaimer)
+            }
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/presenter/SettingPresenter.kt]: This is the bit that does the work of this patch.

File: app/src/main/java/mozilla/lockbox/store/SettingStore.kt [hunk 6]
```diff
@@ -128,19 +134,54 @@ open class SettingStore(
 
         unlockWithFingerprintPendingAuth = rxPrefs.getBoolean(Keys.UNLOCK_WITH_FINGERPRINT_PENDING_AUTH).asObservable()
 
+        val defaultAutoLockTime =
+            if (fingerprintStore.isDeviceSecure) Constant.SettingDefault.autoLockTime else Constant.SettingDefault.noSecurityAutoLockTime
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/store/SettingStore.kt]: 👏

File: app/src/test/java/mozilla/lockbox/presenter/LockedPresenterTest.kt [hunk 7]
```diff
@@ -22,6 +22,7 @@ import mozilla.lockbox.store.LockedStore
 import mozilla.lockbox.store.SettingStore
 import org.junit.Assert
 import org.junit.Before
+import org.junit.Ignore
```
Comment (jhugman) [inline app/src/test/java/mozilla/lockbox/presenter/LockedPresenterTest.kt]: TIL

File: app/src/main/java/mozilla/lockbox/action/RouteAction.kt [hunk 8]
```diff
@@ -55,53 +39,6 @@ sealed class RouteAction(
...
-            Constant.Privacy.uri,
-            R.string.privacy,
-            TelemetryEventObject.settings_faq)
-
-        object SendFeedback : AppWebPage(
-            Constant.SendFeedback.uri,
-            R.string.nav_menu_feedback,
-            TelemetryEventObject.settings_provide_feedback)
-    }
-
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/action/RouteAction.kt]: Oh, I see.

**Pull request merged**